components/basin/sources/new-submission-received/new-submission-received.mjs

```diff
@@ -0,0 +1,69 @@
+import basin from "../../basin.app.mjs";
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+
+export default {
+  name: "New Submission Received",
+  version: "0.0.1",
+  key: "basin-new-submission-received",
+  description: "Emit new event on each new submission received.",
+  type: "source",
+  dedupe: "unique",
+  props: {
+    basin,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      static: {
+        intervalSeconds: DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+      },
+    },
+  },
+  methods: {
+    emitEvent(data) {
+      this.$emit(data, {
+        id: data.id,
+        summary: `New submission received with ID ${data.id}`,
+        ts: Date.parse(data.created_at),
+      });
+    },
+    _setLastSyncDate(date) {
+      this.db.set("lastSyncDate", date);
+    },
+    _getLastSyncDate() {
+      return this.db.get("lastSyncDate");
+    },
+  },
+  hooks: {
+    async deploy() {
+      const { submissions } = await this.basin.getSubmissions();
+
+      submissions.slice(-10).reverse()
+        .forEach(this.emitEvent);
+    },
+  },
+  async run() {
+    const lastSyncDate = this._getLastSyncDate() ?? (new Date).getTime();
+    this._setLastSyncDate((new Date).getTime());
```

Comment:
The `lastSyncDate` is not the same of `(new Date).getTime()`.

Reply:
Don't have difference between store the latest date or the run date, because if don't had any new event on the run, the next events will be mandatorily after the run date.